src/mpid/ch4/src/ch4r_win.c

```diff
@@ -517,31 +510,34 @@ static int win_shm_alloc_impl(MPI_Aint size, int disp_unit, MPIR_Comm * comm_ptr
 
     /* first try global symmetric heap segment allocation */
     if (global_symheap_flag) {
+        size_t my_offset = (shm_comm_ptr) ? shm_offsets[shm_comm_ptr->rank] : 0;
         MPIDIG_WIN(win, mmap_sz) = mapsize;
-        mpi_errno = MPIDIU_get_shm_symheap(mapsize, shm_offsets, comm_ptr,
-                                           win, &symheap_mapfail_flag);
-        if (mpi_errno != MPI_SUCCESS)
+        mpi_errno =
+            MPIDU_shm_alloc_symm_all(comm_ptr, mapsize, my_offset, &MPIDIG_WIN(win, mmap_addr));
+        if (mpi_errno == MPI_ERR_OTHER)
             goto fn_fail;
```

Comment:
Can `mpi_errno` be another value (i.e., not `MPI_ERR_OTHER` and not `MPIDU_ERR_SYM_SHMEM_FAIL`)?